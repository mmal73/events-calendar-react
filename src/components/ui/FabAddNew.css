.fab-add{
    border-radius: 100%;
    bottom: 3rem;
    right: 2rem;
    font-size: 1.6rem;
    position: fixed;
    z-index: 10;
}
.fab-delete{
    border-radius: 100%;
    bottom: 8rem;
    right: 2rem;
    font-size: 1.6rem;
    position: fixed;
    z-index: 10;
}