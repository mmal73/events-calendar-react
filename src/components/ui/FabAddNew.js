import React from 'react';
import './FabAddNew.css';

export default function FabAddNew({handleOpenModal}) {
    return (
        <button
            className="btn btn-primary fab p-3"
            onClick={handleOpenModal}
        >
            <i className="fas fa-plus"></i>
        </button>
    )
}
