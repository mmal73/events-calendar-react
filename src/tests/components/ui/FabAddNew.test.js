import { React } from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import FabAddNew from '../../../components/ui/FabAddNew';

describe('Test in <FabAddNew/> component', () => {
    test('should show text from props', () => {
        const iconText = <i className="fas fa-plus"></i>;
        const handleOpenModal = () => {
            return 'Hola mundo';
        };
        const className = 'btn-primary fab-add';
        const wrapper = shallow(
            <FabAddNew
                className={className}
                text={iconText}
                handleClick={handleOpenModal}
            />
        );

        expect( wrapper ).toMatchSnapshot();
    });
    
})
