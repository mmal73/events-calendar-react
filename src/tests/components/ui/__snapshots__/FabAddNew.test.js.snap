// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <FabAddNew/> component should show text from props 1`] = `
<button
  className="btn btn-primary fab-add"
  onClick={[Function]}
>
  <i
    className="fas fa-plus"
  />
</button>
`;
